### PR TITLE
Product list

### PR DESCRIPTION
**Pull request by smrizzo**
Finished Up Product List since didn't get a change to complete last sprint. 

Pull Request include: 

1. Is now able to add product from product list page which will update the list on the fly as well as top products in artisan detail.

2. Also made so can tap a product in list and navigates to product detail page

3. Fixed edit artisan bug where wasn't updating after visiting second time for same artisan.

4. Fixed delete bugs 

5. Also a lot of semi-colon stuff gong on with linter

### CODE REVIEW

**Review by mpjimenez — APPROVED** [review]
Tested this only on IOS. Everything works the way it should.

**Comment by mpjimenez** [comment]
From the 'ViewAll' page we can successfully get to the ProductDetail page. 
Delete artisan works as well.

File: src/screens/Products/ProductList.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+          <Text style={styles.text}>{`Price: $${item.StandardPrice}`}</Text>
+          <Text style={styles.text}>{`Quantity: ${item.Quantity}`}</Text>
+        </View>
+      </TouchableOpacity>
+    )
+  }
+
+  _keyExtractor = (item) => item.productID
+
+
```
Comment (smrizzo) [inline src/screens/Products/ProductList.js:70]: Probably don't need the extra spacing between these two methods. One spacing is good

File: src/screens/Products/actions.js [hunk 2]
```diff
@@ -18,17 +18,17 @@ export function createProduct(data, artisanID) {
...
+        Title: data.title,
+        Description: data.description,
+        Gender: data.gender,
+        StandardPrice: data.standardPrice,
+        SellerSKU: data.sellerSKU,
+        Quantity: data.quantity,
+        ProductionTime: data.productionTime,
+        TimesSold: data.timesSold,
+        ProductID: db_ref.key
       }
```
Comment (smrizzo) [inline src/screens/Products/actions.js:32]: Glad this was changed so we can be consistent with naming from database

File: src/screens/Products/index.js [hunk 3]
```diff
@@ -19,16 +19,11 @@ function mapDispatchToProps(dispatch) {
...
+const ProductList = connect(mapStateToProps, mapDispatchToProps)(UnconnectedProductList)
+const ProductDetail = connect(mapDispatchToProps,mapDispatchToProps)(UnconnedtedProductDetail)
 
-const ProductDetail = connect(
-  mapDispatchToProps,
-  mapDispatchToProps
-)(UnconnedtedProductDetail)
-
-export { AddProduct, ProductDetail }
+export { AddProduct, ProductDetail, ProductList }
```
Comment (smrizzo) [inline src/screens/Products/index.js:29]: Overall looks pretty good. I see no inline styling is present so good job and changing that.

File: src/screens/Products/index.js [hunk 4]
```diff
@@ -19,16 +19,11 @@ function mapDispatchToProps(dispatch) {
...
 
 import { default as UnconnectedAddProduct } from './AddProduct'
+import { default as UnconnectedProductList } from './ProductList'
 import { default as UnconnedtedProductDetail } from './ProductDetail'
 
-const AddProduct = connect(
-  mapStateToProps,
-  mapDispatchToProps
-)(UnconnectedAddProduct)
+const AddProduct = connect(mapStateToProps, mapDispatchToProps)(UnconnectedAddProduct)
```
Comment (smrizzo) [inline src/screens/Products/index.js:25]: Also good job and changing this since we decided to keep imports on one line until needs to be wrapped.

**Review by smrizzo — COMMENTED** [review]
Overall looks good and seems to have had improvements made based on feedback from other code reviews.

**Pull request merged**